### PR TITLE
added query getRecentComments

### PR DESCRIPTION
**Pull request by RevylaV**
Знімок екрана 2021-03-14 о 08 35 03

### CODE REVIEW

File: modules/comment/comment.permissions.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const commentPermissionsQuery = {
   getCommentById: allow,
   getAllCommentsByProduct: allow,
   getAllCommentsByUser: allow,
+  getRecentComments: allow,
```
Comment (AndriyLymych) [inline modules/comment/comment.permissions.js:10]: please, add permissions just for admin and superadmin

**Comment by Kiyooooko** [comment]
please use our PR template when you adding a PR

File: modules/comment/comment.resolver.js [hunk 2]
```diff
@@ -16,6 +16,19 @@ const commentsQuery = {
     }
   },
 
+  getRecentComments: async (parent, args) => {
+    try {
+      return commentsService.getRecentComments(args.comentDate);
+    } catch (error) {
+      return [
```
Comment (AndriyLymych) [inline modules/comment/comment.resolver.js:23]: we need to return RuleError

File: modules/comment/comment.service.js [hunk 3]
```diff
@@ -49,6 +50,18 @@ class CommentsService extends FilterHelper {
     return comment;
   }
 
+  async getRecentComments({ id, comentDate }) {
+    const comments = await Comment.findById(id).exec();
+    if (!comments) {
+      throw new Error(COMMENT_NOT_FOUND);
+    }
```
Comment (AndriyLymych) [inline modules/comment/comment.service.js:57]: we don't need  comment by id

File: modules/comment/comment.resolver.js (hunk 2 above)
Comment (Kiyooooko) [reply modules/comment/comment.resolver.js:23]: in RuleError please use error.message and error.statusCode

File: modules/comment/comment.service.js [hunk 4]
```diff
@@ -49,6 +50,18 @@ class CommentsService extends FilterHelper {
...
+    const comment = await Comment.find({ comments: comentDate }).exec();
```
Comment (Kiyooooko) [inline modules/comment/comment.service.js:58]: how are going to find comments by this commentDate?

File: modules/comment/comment.service.js [hunk 5]
```diff
@@ -49,6 +50,18 @@ class CommentsService extends FilterHelper {
...
+    if (!comment) {
+      throw new Error(COMMENT_NOT_FOUND);
```
Comment (Kiyooooko) [inline modules/comment/comment.service.js:60]: please use RuleError

File: modules/comment/comment.resolver.js [hunk 6]
```diff
@@ -16,6 +16,19 @@ const commentsQuery = {
     }
   },
 
+  getRecentComments: async (parent, args) => {
```
Comment (Kiyooooko) [inline modules/comment/comment.resolver.js:19]: why you're using async without await?

File: modules/comment/comment.service.js [hunk 7]
```diff
@@ -49,6 +50,16 @@ class CommentsService extends FilterHelper {
     return comment;
   }
 
+  async getRecentComments({ limit, skip }) {
+    const comments = await Comment.find()
+      .sort({ date: -1 })
+      .exec();
+    if (!comments) {
+      throw new Error(COMMENT_NOT_FOUND);
```
Comment (AndriyLymych) [inline modules/comment/comment.service.js:58]: USE RULE ERROR!and also put statuse code

File: modules/comment/comment.service.js [hunk 8]
```diff
@@ -49,6 +51,18 @@ class CommentsService extends FilterHelper {
...
   }
 
+  async getRecentComments(limit) {
+    const comments = await Comment.find()
+      .sort({ date: -1 })
+      .limit(limit)
+      .exec();
+
+    if (!comments.length) {
+      throw new Error(COMMENTS_NOT_FOUND);
```
Comment (AndriyLymych) [inline modules/comment/comment.service.js:61]: use RuleError

File: modules/comment/comment.resolver.js [hunk 9]
```diff
@@ -16,6 +17,14 @@ const commentsQuery = {
     }
   },
 
+  getRecentComments: (_, { limit }) => {
+    try {
+      return commentsService.getRecentComments(limit);
```
Comment (Kiyooooko) [inline modules/comment/comment.resolver.js:22]: please use await, if you use try{}catch(){}

File: modules/comment/comment.resolver.js [hunk 10]
```diff
@@ -16,6 +17,14 @@ const commentsQuery = {
     }
   },
 
+  getRecentComments: (_, { limit }) => {
```
Comment (Kiyooooko) [inline modules/comment/comment.resolver.js:20]: please use async, if you use try{}catch(){}

File: modules/comment/comment.service.js [hunk 11]
```diff
@@ -49,6 +52,18 @@ class CommentsService extends FilterHelper {
...
+  async getRecentComments(limit) {
+    try {
+      const comments = await Comment.find()
+        .sort({ date: -1 })
+        .limit(limit)
+        .exec();
+      return comments;
+    } catch (error) {
+      throw new RuleError(error.message, error.statusCode);
+    }
```
Comment (AndriyLymych) [inline modules/comment/comment.service.js:64]: you need do something like that...without try..catch here
image

**Review by AndriyLymych — DISMISSED** [review]

**Comment by Kiyooooko** [comment]
pls solve conflicts
Screenshot 2021-03-23 at 18 32 48

**Review by Haytham1337 — APPROVED** [review]

**Review by MMashivskij — APPROVED** [review]

**Pull request merged**